src/identity/core/getName.test.tsx

```diff
@@ -52,4 +59,67 @@ describe('getName', () => {
     const name = await getName({ address: walletAddress });
     expect(name).toBe(null);
   });
+
+  it('should return mainnet username', async () => {
+    const walletAddress =
+      '0x1234567890123456789012345678901234567890' as Address;
+    const expectedEnsName = 'leo.eth';
+    mockGetEnsName.mockResolvedValue(expectedEnsName);
+    const name = await getName({ address: walletAddress, chainId: mainnet.id });
+    expect(name).toBe(expectedEnsName);
+    expect(getChainPublicClient).toHaveBeenCalledWith(mainnet.id);
+  });
+
+  it('should return sepolia username', async () => {
+    const walletAddress =
+      '0x1234567890123456789012345678901234567890' as Address;
+    const expectedEnsName = 'leo.test.eth';
+    mockGetEnsName.mockResolvedValue(expectedEnsName);
+    const name = await getName({ address: walletAddress, chainId: sepolia.id });
+    expect(name).toBe(expectedEnsName);
+    expect(getChainPublicClient).toHaveBeenCalledWith(sepolia.id);
+  });
+
+  it('should return custom testnet chain username', async () => {
+    const walletAddress =
+      '0x1234567890123456789012345678901234567890' as Address;
+    const expectedEnsName = 'leo.customtestnet.eth';
+    mockReadContract.mockResolvedValue(expectedEnsName);
+    const name = await getName({
+      address: walletAddress,
+      chainId: baseSepolia.id,
+    });
+    expect(name).toBe(expectedEnsName);
+    expect(getChainPublicClient).toHaveBeenCalledWith(baseSepolia.id);
+  });
+
+  it('should return custom mainnet username', async () => {
+    const walletAddress =
+      '0x1234567890123456789012345678901234567890' as Address;
+    const expectedEnsName = 'leo.custommainnet.eth';
+    mockReadContract.mockResolvedValue(expectedEnsName);
+    const name = await getName({ address: walletAddress, chainId: base.id });
+    expect(name).toBe(expectedEnsName);
+    expect(getChainPublicClient).toHaveBeenCalledWith(base.id);
+  });
+
+  it('should return null if user is not registered', async () => {
+    const walletAddress =
+      '0x1234567890123456789012345678901234567890' as Address;
+    const expectedEnsName = null;
+    mockReadContract.mockResolvedValue(expectedEnsName);
+    const name = await getName({ address: walletAddress, chainId: base.id });
+    expect(name).toBe(expectedEnsName);
+    expect(getChainPublicClient).toHaveBeenCalledWith(base.id);
+  });
+
+  it('should throw an error on unsupported chain', async () => {
+    const walletAddress =
```

Comment:
As walletAddress is always the same, you can declare that at the top of the describe